lib/twilio/device.ts

```diff
@@ -1310,8 +1317,11 @@ class Device extends EventEmitter {
     this._audio = new (this._options.AudioHelper || AudioHelper)(
       this._updateSinkIds,
       this._updateInputStream,
-      getUserMedia,
-      { audioContext: Device.audioContext },
+      this._options.getUserMedia || getUserMedia,
+      {
+        audioContext: Device.audioContext,
+        enumerateDevices: this._options.enumerateDevices,
```

Comment:
I agree it's not very clear. But `gUM` is a required param of AudioHelper though. So if the override is not provided, we provide a default value. But `enumerateDevices` here is optional, hence, putting it inside the `AudioHelper.Options` object.

Reply:
Sorry, my thought process isn't clear here. It's more like sometimes we cache these import interceptions in the constructor within the `this._options` member, but sometimes we don't and we do this "use the option, fallback to the import" logic in different places. Not something to tackle now, but it should all be in the constructor in my opinion. And if we need to propagate it, then we shouldn't have to do any more logic than simply `{ gum: this._options.gUM }` or something like that.

Reply:
Got it and I agree. I have the same observation. But like you said, we don't want to touch all those areas that are not related to this PR. Having said that, we can probably do something better here, but it still not clear what your suggestion is for this specific area of the code. Maybe type out your suggestion?

Reply:
Something like `this._options.gUM = options.gUM || import.gUM` and here we would just use `this._options.gUM`. If it's `undefined` at this point, the consuming class (`Call` in this case) would have it's own logic to handle this.

Reply:
I see. In this case. `this._options` does not mirror `options` anymore since `this._options.gUM` will always be available. But I get what you mean by `caching` now. Instead of saving it inside `this._options`, maybe another member something like `this._getUserMedia = options.getUserMedia || import.gum`? But then again, we don't really gain much since we already 'cache' the options. I'm leaning more towards keeping it this way. If you look at other options in this class, we follow a similar pattern.

Going back to your original comment
>It's more like sometimes we cache these import interceptions in the constructor within the this._options member, but sometimes we don't and we do this "use the option, fallback to the import" logic in different places.

I look at the device class and we don't really cache the import interceptions (unless I missed it?) or maybe we're doing it on other files? Probably part of a future discussion maybe?